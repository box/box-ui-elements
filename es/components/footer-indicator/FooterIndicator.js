import * as React from 'react';
import Tooltip, { TooltipPosition } from '../tooltip';
import IconPuzzlePiece from '../../icons/general/IconPuzzlePiece';
import './FooterIndicator.scss';
const FooterIndicator = ({
  indicatorText
}) => {
  return /*#__PURE__*/React.createElement("div", {
    className: "bdl-FooterIndicator"
  }, /*#__PURE__*/React.createElement(Tooltip, {
    targetWrapperClassName: "bdl-FooterIndicator-tooltipTarget",
    position: TooltipPosition.TOP_RIGHT,
    text: indicatorText
  }, /*#__PURE__*/React.createElement("div", {
    className: "bdl-FooterIndicator-content"
  }, /*#__PURE__*/React.createElement("span", {
    className: "bdl-FooterIndicator-iconWrapper"
  }, /*#__PURE__*/React.createElement(IconPuzzlePiece, {
    height: 14,
    width: 14
  })), /*#__PURE__*/React.createElement("span", {
    className: "bdl-FooterIndicator-text"
  }, indicatorText))));
};
export default FooterIndicator;
//# sourceMappingURL=FooterIndicator.js.map