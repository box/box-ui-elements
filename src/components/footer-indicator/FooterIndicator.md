### Description

Renders a small tab with text in the bottom-left corner of a page.

### Examples

#### Default

```
<div style={{
    height: '250px',
    position: 'relative',
    transform: 'translate3d(0,0,0)',
}}>
    <FooterIndicator
        indicatorText="FooterIndicator"
    />
</div>
```

#### With truncated text

```
<div style={{
    height: '250px',
    position: 'relative',
    transform: 'translate3d(0,0,0)',
}}>
    <FooterIndicator
        indicatorText="FooterIndicatorWithExtremelyRemarkablyStupendouslyTerrificallyLongName"
    />
</div>
```

#### In a `NavSidebar`

```js
const NavSidebar = require('box-ui-elements/es/components/nav-sidebar/NavSidebar')
  .default;
const NavList = require('box-ui-elements/es/components/nav-sidebar/NavList')
  .default;

<div
  style={{
    transform: 'translate3d(0,0,0)',
  }}
>
  <NavSidebar style={{ padding: 0 }}>
    <div style={{ padding: '20px' }}>
      <NavList>
        <Link>Item A</Link>
        <Link>Item B</Link>
        <Link>Item C</Link>
        <Link>Item D</Link>
        <Link>Item E</Link>
      </NavList>
    </div>
    <FooterIndicator indicatorText="FooterIndicatorInANavSidebar" />
  </NavSidebar>
</div>;
```

#### In a `LeftSidebar`

```js
const LeftSidebar = require('box-ui-elements/es/features/left-sidebar/LeftSidebar')
  .default;
const IconReturnToAdminConsole = require('box-ui-elements/es/features/left-sidebar/icons/IconReturnToAdminConsole')
  .default;
const IconFilePDF = require('../../icons/file/IconFilePDF').default;

const leftSidebarProps = {
  customTheme: {
    isLight: 'yes',
    primaryColorLight: '#f0f0f0', // selected menu item background color
    primaryColorLighter: '#123',
    primaryColorDark: '#123',
    primaryColorDarker: '#123',
    contrastColor: '#123',
    secondaryColor: '#123', // icons, selected menu item border, selected menu item icon
  },
  htmlAttributes: {
    'data-resin-component': 'leftnav',
  },
  copyrightFooterProps: {
    'data-resin-target': 'copyright',
    href: '/about-us',
  },
  indicatorText: 'FooterIndicatorInALeftSidebar',
  instantLoginProps: {
    htmlAttributes: {
      href: '/logout',
      'data-resin-target': 'exitinstantlogin',
      rel: 'external',
    },
    iconComponent: IconReturnToAdminConsole,
    id: 'return-to-admin-console',
    message: 'Return to Admin Console',
    showTooltip: true,
  },
  isInstantLoggedIn: true,
  isDragging: true,
};

const menuItems = [
  {
    id: 'favorites',
    canReceiveDrop: false,
    message: 'Favorites',
    htmlAttributes: {
      href: '/favorites',
      rel: 'some-rel',
      target: '_blank',
      'data-resin-target': 'favorites',
    },
    selected: false,
    showTooltip: true,
    showLoadingIndicator: false,
    placeholder: 'Drag items here for quick access',
    menuItems: [
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-1',
        message: 'Some Really Longggggg Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-2',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-3',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-4',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-5',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-6',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-7',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-8',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-9',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-10',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-11',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-12',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-13',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-14',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-15',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-16',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-17',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-18',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
      {
        iconComponent: IconFilePDF,
        id: 'favorites-item-19',
        message: 'Some Short Favorite',
        htmlAttributes: {
          href: '/1234',
          'data-resin-feature': 'favorites',
          'data-resin-target': 'itemname',
        },
        scaleIcon: true,
        showTooltip: true,
      },
    ],
  },
];

<div
  style={{
    transform: 'translate3d(0,0,0)',
  }}
>
  <LeftSidebar menuItems={menuItems} leftSidebarProps={leftSidebarProps} />
</div>;
```
