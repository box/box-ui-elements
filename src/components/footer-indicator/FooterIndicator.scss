@import '../../styles/variables';

.bdl-FooterIndicator {
    position: fixed;
    bottom: 0;
    z-index: 10;
    width: 170px;
    height: 32px;
    margin: 0 15px;
    background: $bdl-gray-80;
    border-radius: $bdl-border-radius-size $bdl-border-radius-size 0 0;
}

.bdl-FooterIndicator-content {
    display: flex;
    align-items: center;
    justify-content: flex-start;
    height: 100%;
    padding: 10px;

    &:focus {
        outline: none;
    }
}

.bdl-FooterIndicator-iconWrapper {
    display: flex;
    margin-right: 9px;
}

.bdl-FooterIndicator-text {
    overflow: hidden;
    color: $white;
    font-size: 11px;
    white-space: nowrap;
    text-overflow: ellipsis;
}
