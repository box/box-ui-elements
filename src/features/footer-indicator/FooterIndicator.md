### Description

Renders a small tab with text in the bottom-left corner of a page.

### Examples

#### Standard

```
<div style={{
    height: '250px',
    position: 'relative',
    transform: 'translate3d(0,0,0)',
}}>
    <FooterIndicator
        indicatorText="FooterIndicator"
    />
</div>
```

#### With text overflow

```
<div style={{
    height: '250px',
    position: 'relative',
    transform: 'translate3d(0,0,0)',
}}>
    <FooterIndicator
        indicatorText="FooterIndicatorWithExtremelyLongName"
    />
</div>
```
