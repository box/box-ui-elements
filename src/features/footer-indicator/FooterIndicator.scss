@import '../../styles/variables';
@import '../left-sidebar/styles/_sizes.media';

.bdl-FooterIndicator {
    background: $fours;
    border-radius: 4px 4px 0 0;
    bottom: 0;
    height: 32px;
    margin: 0 15px;
    position: fixed;
    width: 170px;
    z-index: 10;
}

.bdl-FooterIndicator-content {
    display: flex;
    justify-content: flex-start;

    &:focus {
        outline: none;
    }
}

.bdl-FooterIndicator-icon {
    margin: 8px 10px;
}

.bdl-FooterIndicator-text {
    color: $white;
    font-size: 11px;
    margin: 6px 10px 6px 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;

    &::after {
        content: '';
        display: block;
    }
}

// Override default tooltip visibility
.bdl-FooterIndicator-tooltip {
    display: none;

    &.bdl-is-visible {
        display: block;
    }
}
