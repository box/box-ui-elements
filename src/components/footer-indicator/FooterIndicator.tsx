import * as React from 'react';
import Tooltip, { TooltipPosition } from '../tooltip';
import IconPuzzlePiece from '../../icons/general/IconPuzzlePiece';

import './FooterIndicator.scss';

type Props = {
    indicatorText: string;
};

const FooterIndicator = ({ indicatorText }: Props) => {
    return (
        <div className="bdl-FooterIndicator">
            <Tooltip position={TooltipPosition.TOP_RIGHT} text={indicatorText}>
                <div className="bdl-FooterIndicator-content">
                    <span className="bdl-FooterIndicator-iconWrapper">
                        <IconPuzzlePiece height={14} width={14} />
                    </span>
                    <span className="bdl-FooterIndicator-text">{indicatorText}</span>
                </div>
            </Tooltip>
        </div>
    );
};

export default FooterIndicator;
