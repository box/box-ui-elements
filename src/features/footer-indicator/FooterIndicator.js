// @flow
import * as React from 'react';
import classNames from 'classnames';

import Tooltip from '../../components/tooltip';

import IconPuzzlePiece from '../../icons/general/IconPuzzlePiece';

import './FooterIndicator.scss';

type Props = {
    indicatorText: string,
};

type State = {
    isTextOverflowed: boolean,
};

class FooterIndicator extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {
            isTextOverflowed: false,
        };
    }

    componentDidMount() {
        if (!this.indicatorTextElement) {
            return;
        }

        const { offsetWidth, scrollWidth } = this.indicatorTextElement;

        if (offsetWidth < scrollWidth) {
            // eslint-disable-next-line react/no-did-mount-set-state
            this.setState({
                isTextOverflowed: true,
            });
        }
    }

    indicatorTextElement: ?HTMLElement;

    render() {
        const { indicatorText } = this.props;
        const { isTextOverflowed } = this.state;
        return (
            <div className="bdl-FooterIndicator">
                <Tooltip
                    className={classNames('bdl-FooterIndicator-tooltip', {
                        'bdl-is-visible': isTextOverflowed,
                    })}
                    position="middle-right"
                    text={indicatorText}
                >
                    <div className="bdl-FooterIndicator-content">
                        <IconPuzzlePiece className="bdl-FooterIndicator-icon" />
                        <span
                            ref={textElement => {
                                this.indicatorTextElement = textElement;
                            }}
                            className="bdl-FooterIndicator-text"
                        >
                            {indicatorText}
                        </span>
                    </div>
                </Tooltip>
            </div>
        );
    }
}

export default FooterIndicator;
