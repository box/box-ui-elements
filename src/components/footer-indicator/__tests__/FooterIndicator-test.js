import * as React from 'react';

import FooterIndicator from '../FooterIndicator';

describe('feature/footer-indicator/FooterIndicator', () => {
    const getWrapper = () => shallow(<FooterIndicator indicatorText="abcdefghijklmnopqrstuvwxyz" />);

    test('should render a FooterIndicator', () => {
        const wrapper = getWrapper();

        expect(wrapper).toMatchSnapshot();
    });

    test('should render a tooltip if isTextOverflowed is true', () => {
        const isTextOverflowed = true;
        const wrapper = getWrapper();
        wrapper.setState({ isTextOverflowed });

        expect(wrapper).toMatchSnapshot();
    });

    test('should not render a tooltip if isTextOverflowed is false', () => {
        const isTextOverflowed = false;
        const wrapper = getWrapper();
        wrapper.setState({ isTextOverflowed });

        expect(wrapper).toMatchSnapshot();
    });
});
