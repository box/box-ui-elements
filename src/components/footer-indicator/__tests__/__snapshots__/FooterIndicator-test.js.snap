// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature/footer-indicator/FooterIndicator should not render a tooltip if isTextOverflowed is false 1`] = `
<div
  className="bdl-FooterIndicator"
>
  <Tooltip
    className="bdl-FooterIndicator-tooltip"
    constrainToScrollParent={false}
    constrainToWindow={true}
    position="middle-right"
    text="abcdefghijklmnopqrstuvwxyz"
    theme="default"
  >
    <div
      className="bdl-FooterIndicator-content"
    >
      <span
        className="bdl-FooterIndicator-iconWrapper"
      >
        <IconPuzzlePiece />
      </span>
      <span
        className="bdl-FooterIndicator-text"
      >
        abcdefghijklmnopqrstuvwxyz
      </span>
    </div>
  </Tooltip>
</div>
`;

exports[`feature/footer-indicator/FooterIndicator should render a FooterIndicator 1`] = `
<div
  className="bdl-FooterIndicator"
>
  <Tooltip
    className="bdl-FooterIndicator-tooltip"
    constrainToScrollParent={false}
    constrainToWindow={true}
    position="middle-right"
    text="abcdefghijklmnopqrstuvwxyz"
    theme="default"
  >
    <div
      className="bdl-FooterIndicator-content"
    >
      <span
        className="bdl-FooterIndicator-iconWrapper"
      >
        <IconPuzzlePiece />
      </span>
      <span
        className="bdl-FooterIndicator-text"
      >
        abcdefghijklmnopqrstuvwxyz
      </span>
    </div>
  </Tooltip>
</div>
`;

exports[`feature/footer-indicator/FooterIndicator should render a tooltip if isTextOverflowed is true 1`] = `
<div
  className="bdl-FooterIndicator"
>
  <Tooltip
    className="bdl-FooterIndicator-tooltip bdl-is-visible"
    constrainToScrollParent={false}
    constrainToWindow={true}
    position="middle-right"
    text="abcdefghijklmnopqrstuvwxyz"
    theme="default"
  >
    <div
      className="bdl-FooterIndicator-content"
    >
      <span
        className="bdl-FooterIndicator-iconWrapper"
      >
        <IconPuzzlePiece />
      </span>
      <span
        className="bdl-FooterIndicator-text"
      >
        abcdefghijklmnopqrstuvwxyz
      </span>
    </div>
  </Tooltip>
</div>
`;
